From 57d2cbc84bb1cf36a1a1bd8d4a5735cae46ffe7a Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Tue, 24 Jul 2018 15:03:10 +0800
Subject: [PATCH 159/205] soc: ti: pm33xx: confine the suspend-related codes
 with SUSPEND

This commit comes from:
  git://git.ti.com/processor-sdk/processor-sdk-linux.git

Definitely, suspend_wfi_flags belongs to CONFIG_SUSPEND.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/soc/ti/pm33xx.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/soc/ti/pm33xx.c b/drivers/soc/ti/pm33xx.c
index 44600bd..83101cf 100644
--- a/drivers/soc/ti/pm33xx.c
+++ b/drivers/soc/ti/pm33xx.c
@@ -333,6 +333,7 @@ static int am33xx_pm_probe(struct platform_device *pdev)
 	suspend_set_ops(&am33xx_pm_ops);
 #endif /* CONFIG_SUSPEND */
 
+#ifdef CONFIG_SUSPEND
 	/*
 	 * For a system suspend we must flush the caches, we want
 	 * the DDR in self-refresh, we want to save the context
@@ -343,6 +344,7 @@ static int am33xx_pm_probe(struct platform_device *pdev)
 	suspend_wfi_flags |= WFI_FLAG_SELF_REFRESH;
 	suspend_wfi_flags |= WFI_FLAG_SAVE_EMIF;
 	suspend_wfi_flags |= WFI_FLAG_WAKE_M3;
+#endif /* CONFIG_SUSPEND */
 
 	ret = pm_ops->init(am33xx_do_sram_idle);
 	if (ret) {
-- 
1.7.9.5

